### ISSUE TITLE
(Feature request) Template mixins: Grouped inherited promoted attributes

### ISSUE THREAD

**Issue by mm21** [zadam/trilium#3761]
### Describe feature

I like the idea of designating template notes as "mixins", wherein they define a group of promoted attributes which I can reuse elsewhere in other template notes.

Say I have a template note "Address" with the following attributes:

`#label:street=promoted,single,text #label:city=promoted,single,text #label:state=promoted,single,text #label:zip=promoted,single,text` 

This results in promoted attributes like:

![image](https://user-images.githubusercontent.com/8033134/227357752-b3ef31e4-a084-473b-b2c9-28b3c4aa2a59.png)

But when I attempt to use this as a mixin in another template like "Contact" which is a person with an address:

`#label:firstName=promoted,single,text #label:lastName=promoted,single,text #label:nickname=promoted,single,text ~inherit=Address`

The promoted attributes are interleaved:

![image](https://user-images.githubusercontent.com/8033134/227356300-1c947d05-8681-4021-bc91-8555e1f3280b.png)

But I'd like them to display in the same order they were grouped within the Address "mixin":

![image](https://user-images.githubusercontent.com/8033134/227357481-db2b026f-848d-4a58-9207-229282a9a9bb.png)

I did some digging and see this behavior is intentional (see also https://github.com/zadam/trilium/issues/3368) - the first promoted attribute is displayed from each group (owned or inherited promoted attributes), then the second from each group, etc.

I don't think this needs a whole new "template mixin" feature, but just off the top of my head, maybe an attribute which says how to order promoted attributes. Say #promotedAttributeOrder, with possible values of "interleave" (default, how it is now) and "group" (preserve inherited attribute order within each ~template/~inherit relation target).

If the grouped inherited attributes could also have some kind of border indicating they're all from the same ~template/~inherit relation, like the red border above (using the theme border), it would be a real game changer. (And this app is already a game changer, for the record.)

**Comment by zadam** [comment]
Hello, the interleaving is actually not intended. Fixed.

**Comment by zadam** [comment]
The current solution is certainly not ideal, but IMHO an improvement over previous interleaving.

Relying on `noteId` for sort is arbitrary, but also stable - so you can sort of get used to the order of grouped promoted attributes. 

Defining an order using an algorithm you describe is possible, but more involved, it's not compatible with the current way of collecting attributes / caching them.

**Comment by mm21** [comment]
Sure, that's fair. If I took a stab at it and found a simple enough way to do what I described, would you consider taking a PR for this?

**Comment by zadam** [comment]
Yes, but conditional that the PR is not too complex / disruptive.
